huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go

```diff
@@ -0,0 +1,281 @@
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			// Computed
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"device_owner": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mac_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			// Deprecated
+			"subnet_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Computed:      true,
+				ConflictsWith: []string{"ip_version"},
+				Deprecated:    "use ip_version instead",
+			},
+		},
+	}
+}
+
+func resourceNetworkingVipCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.NetworkingV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud VPC network v1 client: %s", err)
+	}
+
+	networkId := d.Get("network_id").(string)
+	n, err := subnets.Get(client, networkId).Extract()
+	if err != nil {
+		return fmtp.DiagErrorf("Error retrieving HuaweiCloud subnet by network ID (%s): %s", networkId, err)
+	}
+
+	// Check whether the subnet ID entered by the user belongs to the same subnet as the network ID.
+	subnetId := d.Get("subnet_id").(string)
+	if subnetId != "" && subnetId != n.SubnetId && subnetId != n.IPv6SubnetId {
+		return fmtp.DiagErrorf("The subnet ID does not belong to the subnet where the network ID is located.")
+	}
+
+	// Pre-check for subnet network, the virtual IP of IPv6 must be established on the basis that the subnet supports
+	// IPv6.
+	if d.Get("ip_version").(int) == 6 {
+		if n.IPv6SubnetId == "" {
+			return fmtp.DiagErrorf("The subnet does not support IPv6, please enable IPv6 first.")
+		}
+		subnetId = n.IPv6SubnetId
+	} else {
+		subnetId = n.SubnetId
+	}
+
+	opts := ports.CreateOpts{
+		Name:        d.Get("name").(string),
+		DeviceOwner: "neutron:VIP_PORT",
+		NetworkId:   networkId,
+		FixedIps: []ports.FixedIp{
+			{
+				SubnetId:  subnetId,
+				IpAddress: d.Get("ip_address").(string),
+			},
+		},
+	}
+
+	logp.Printf("[DEBUG] Updating network VIP (%s) with options: %#v", d.Id(), opts)
+	vip, err := ports.Create(client, opts)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud network VIP: %s", err)
+	}
+	logp.Printf("[DEBUG] Waiting for HuaweiCloud network VIP (%s) to become available.", vip.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"BUILD"},
+		Target:     []string{"DOWN"},
+		Refresh:    waitForNetworkVipStateRefresh(client, vip.ID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      3 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(vip.ID)
```

Comment:
should set the resource ID after it was created successfully

Reply:
fixed.
Another place where the code for setting the ID has been removed.